<b>Supplementary Data</b>

SData 1.xlsx  List of targets for the RNA sequence capture imprinting study <br> 
SData 2.xlsx  Read count from sequencing and map rates of all reads and informative reads <br> 
SData 3.xlsx  Filtering on genes based on set thresholds <br> 
SData 4.xlsx  SNP vs Informative Read Pipeline (IRP) <br>
SData 5.xlsx  Tissue specific enrichment of transcripts in the seed during seed development <br>
SData 6.xlsx  onservative, Moderate stringency , Low stringency sets for investigation of imprinting <br>
SData 7.xlsx  Preferential expression analysis <br>
SData 8.xlsx  Differential expression analysis between ecotypes used in this study <br>
SData 9.xlsx  Differential expression analysis for all mutant crosses calling parental preferential bias <br>
SData 10.xlsx  Prediction interval analysis of parentally biased genes in mutant crosses <br>

<b>Read Counts</b>

<b>Thresholds </b>



<b>Guide to Additional Data and Program Files on Github</b><br>
(Directories: project1; core; ERCC_Normalize)<br>

SGuideRawData 1-17.xlsx<br>
SGuideProgramFiles 1-14.xlsx

